src/main/java/it/uniroma3/mat/extendedset/utilities/IntList.java

```diff
@@ -7,23 +7,11 @@
  */
 public class IntList
 {
-  private final ArrayList<int[]> baseLists = new ArrayList<int[]>();
-
-  private final int allocateSize;
-
-  private int maxIndex;
+  private static final int ALLOCATION_SIZE = 1024;
```

Comment:
E. g. because it is used in integral division in each `IntList.set()` (to which `IntList.add()` also delegates): https://github.com/metamx/extendedset/pull/10/files/31330598a917ce44162ffa69f4a372e4fbedec3a#diff-fd75b71232d6fb392b6a8a841e048df9L45

JIT compiler (if not javac) replaces division by 1024 (a power of two) with bit shift. Integral devision is very slow.

I was benchmarking this change isolated and there was a noticeable speedup.